keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -115,14 +114,12 @@ This proposal introduces the support of both persistent and ephemeral-style volu
 
 * Support for both persistent and ephemeral volumes
 * Propose a way to clearly distinguish between persistent and ephemeral volumes.
-* Persistent inline CSI volumes must be pre-provisioned and manually de-provisioned.
-* Persistent inline CSI volumes will participate in storage flows including attach, stage, mount, unmount, unstage, and detach.
-* Persistent inline volumes must use the handle of pre-provisioned volumes
-* An ephemeral volume is created and deleted automatically following its pod's lifecycle
-* Inline ephemeral volumes receive a generated volume handle
+* Define how persistent inline CSI volumes will work.
+* Outline how ephemeral volumes are created and deleted.
+* Hightlight the security concerns with the design and how to mitigate them.
 
 ### Inline volumes
-Normally, volume operations can be triggered from PV/PVC or when from a pod spec.  Currently, CSI only supports volumes that originate from PV/PVC. When a volume is embedded inside a pod spec, it will be referred to as `inline` in this document.  Volumes that originates from inline pod specs can support two modes:
+Normally, volume operations can be triggered from PV/PVC or when from a pod spec with reference to an embedded volume definition.  Currently, CSI only supports volumes that originate from PV/PVC. When a volume is embedded inside a pod spec, it will be referred to as `inline` in this document.  Volumes that originate from inline pod specs can support two modes:
```

Comment:
Hmm.... I can see what your trying to say and it makes sense. but this is a little awkward here... Not sure how to fix it...

Here's the pieces and maybe that might help:
We've kind of overloaded the term 'volume'... There is a list entry in the 'pod.spec.volume' that can be thought of as a volume. this is what I think you called an 'embedded volume definition' in the first sentence? There is also 'volume' which is the end result of calling the csi.CreateVolume function, of which a volumeHandle is a pointer too. The second part of the sentence, "volumes that originate" sounds odd under the second definition, as its not creating the volume so nothing is originating....

'volume mounts' rather then 'volumes' may make more sense, but its more then just a mount.... So that may not be right.

"volume requests"?

Reply:
@kfox1111 thanks for pointing this out. I will attempt to clarify with clear delineation between volume spec, volume resource.